File: src/foundation/java/io/karma/pda/foundation/component/Box.java
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

package io.karma.pda.foundation.component;

import io.karma.pda.api.app.component.ComponentType;
import io.karma.pda.api.state.MutableState;
import io.karma.pda.api.state.Synchronize;
import io.karma.peregrine.api.color.Color;
import io.karma.peregrine.api.color.ColorProvider;

import java.util.UUID;

/**
 * @author Alexander Hinze
 * @since 23/04/2024
 */
public class Box extends DefaultContainer {
    @Synchronize
    public final MutableState<ColorProvider> background = MutableState.of(Color.NONE);
    @Synchronize
    public final MutableState<ColorProvider> foreground = MutableState.of(Color.NONE);

    public Box(ComponentType<?> type, UUID id) {
        super(type, id);
    }
}

File: src/composition/kotlin/io/karma/pda/composition/component/Box.kt
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

@file:JvmName("ComponentExtensionsKt")
@file:JvmMultifileClass

package io.karma.pda.composition.component

import io.karma.pda.api.app.component.Container
import io.karma.pda.composition.Composable
import io.karma.pda.composition.Composer
import io.karma.pda.composition.LayoutProps
import io.karma.pda.foundation.component.Box
import io.karma.pda.foundation.component.DefaultComponents

/**
 * @author Alexander Hinze
 * @since 07/06/2024
 */

@Composable
inline fun <C : Container> Composer<C>.Box(
    crossinline layoutProps: LayoutProps,
    localName: String? = null,
    props: @Composable Box.() -> Unit = {},
    content: @Composable Composer<Box>.() -> Unit
) = container(
    type = DefaultComponents.BOX,
    layoutProps = layoutProps,
    localName = localName,
    props = props,
    content = content
)